test/y18n-test.cjs

```diff
@@ -343,6 +343,16 @@ describe('y18n', function () {
 
       i18n.__('meow').should.equal('le meow')
     })
+
+    it('prevents prototype pollution', function () {
+      var i18n = y18n()
+
+      i18n.setLocale('__proto__')
+
+      i18n.updateLocale({ polluted: 'Yes! Its Polluted' })
```

Comment:
after this call, shouldn't `expect(i18n.getLocale()).to.have.property('polluted')` pass as well?

Reply:
@ljharb - before the fix, this may have been the case. However, after running the test suite with your suggested test, the following is returned:

<img width="767" alt="Screenshot 2020-10-22 at 22 05 28" src="https://user-images.githubusercontent.com/55323451/96929900-d1b58300-14b2-11eb-8336-333d9d656a0c.png">

Reply:
No, it would suggest that the locale object was properly updated.

I think the solution here isn't to disrupt usage of a string `__proto__`, but instead, to "salt" each locale name with a character that couldn't conflict with anything built-in. I usually use `$`.

ie, every get and set of a locale name should quietly stick a "$" in front. That way it'll be impossible to collide with the name of any builtin.

Reply:
@ljharb - okay, we will take another look at this, and see if we can salt each locale name as suggested with a non-disruptive character.

Cheers! 🍰